src/main/java/software/amazon/event/kafkaconnector/EventBridgeWriter.java

```diff
@@ -180,75 +205,58 @@ public void shutDownEventBridgeClient() {
    * @param record Record to be sent to EventBridge
    * @return the PutEventsRequestEntry ready to send
    */
-  private Optional<PutEventsRequestEntry> createPutEventsEntry(EventBridgeWriterRecord record) {
+  private Either<EventBridgeWriterErrorSinkRecord, Pair<PutEventsRequestEntry, SinkRecord>>
+      createPutEventsEntry(SinkRecord record) {
     try {
-      return Optional.of(
-          PutEventsRequestEntry.builder()
-              .eventBusName(config.eventBusArn)
-              .source(sourcePrefix + config.connectorId)
-              .detailType(config.getDetailType(record.getTopic()))
-              .resources(config.resources)
-              .detail(createJsonPayload(record.getSinkRecord()))
-              .build());
+      return Either.right(
+          Pair.of(
```

Comment:
Creating a pair of PutEventsRequestEntry + SinkRecord sounds very similar to create a Sinkrecord + Error. For one we have a wrapper class, for the other we use a pair. This relates to my comment above. Maybe we can create a generic or specific wrapper classes for SinkRecord + T.

Reply:
You got me. That's one point I'm not happy about. Do you have a good name for the wrapper class? I didn't have one yesterday.

Reply:
EventBridgeSinkRecord or the good old EventBridgeWriterRecord?

Reply:
For a generic wrapper class `SinkRecord` + `T` this came to my mind
```
class SinkRecordMapped<T> {

  ...

  public SinkRecord getSinkRecord() {...}

  public T getValue() {...}

}
```